src/Main.hs

```diff
@@ -37,16 +37,41 @@ import Text.Pandoc.Builder qualified as B
 import Text.Pandoc.Definition (Pandoc (..))
 import Text.Pandoc.Walk qualified as W
 
+import Data.Generics.Sum.Any (AsAny (_As))
+import Data.SOP (I (..), NP (Nil, (:*)))
+import Ema
+import Ema.CLI qualified
+import Ema.Route.Encoder (htmlSuffixPrism, mkRouteEncoder)
+import Ema.Route.Generic
+import Generics.SOP qualified as SOP
+
+main :: IO ()
+main =
+  Ema.runSite_ @Route "./content"
+
 -- ------------------------
 -- Our site route
 -- ------------------------
 
+data Route
+  = Route_Markdown MarkdownRoute
+  | Route_Static StaticRoute
+  deriving stock (Eq, Show, Ord, Generic)
+  deriving anyclass (SOP.Generic, SOP.HasDatatypeInfo)
+  deriving
+    (HasSubRoutes)
+    via ( Route
+            `WithSubRoutes` '[ SlugListRoute
+                             , StaticRoute
+                             ]
+        )
+  deriving (IsRoute) via (Route `WithModel` Model)
+
+instance HasSubModels Route where
+  subModels m =
+    I (Set.map coerce $ Map.keysSet $ modelDocs m) :* I (modelFiles m) :* Nil
```

Comment:
By the way, @lucasvreis - note here that the first sub-model (for markdown routes) is not really a sub-model (ie., it is not merely a field of the larger model) ... but rather a *computation* (`Set.map coerce . Map.keysSet`) derived from that field.

Reply:
Well, you could define `SlugMapRoute` on top of `SlugListRoute`, and this arbitary computation would then belong in `SlugMapRoute`